hangman_game.ex

```diff
@@ -0,0 +1,91 @@
+defmodule Hangman.Game do
+  defmodule State do
+    defstruct(game_state: ":initializing", 
+              turns_left: 7,               
+              shown_letters: [],        #If user won or lost, letters list would be shown_letters list
+              letters: [],
+              used: [],
+              word: "",
+              last_guess: "" 
+              )
+  end
+
+  def tally(state = %State{letters: l, shown_letters: sl}) when (l == sl) do
```

Comment:
It's an interesting approach, testing for a win or a loss in the `tally()` function. 

### -1 idiomatic

However, this is better written as:

~~~ elixir
def tally(state = %State{letters: letters, shown_letters: letters}) do
~~~